### DISCUSSION TITLE
Calculation crashed on postprocessing of anphon

### DISCUSSION THREAD

**Discussion by umichwbaek** [ttadano/alamode#215] [Q&A]
Dear Dr. Tadano,

I got job crash at postprocessing step while running anphon on SCPH mode.

''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
 
Diagonalizing dynamical matrices for all k points ... done !
 Preparing for calculating V4  ... done!
 RESTART_SCPH is true.
 Dynamical matrix is read from file ... done.

 Running postprocess of SCPH/QHA (calculation of free energy, MSD, DOS)
 The number of temperature points:   21

BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
RANK 0 PID 2538917 RUNNING AT c456-227
KILLED BY SIGNAL: 11 (Segmentation fault)

'''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''

It shows 'free(): invalid pointer' error.

It seems the problem is the memory allocation problem. I run the calculation in HPC with 128 GB per node. I also tried multinode calculation up to 10 nodes but it did not work, either.

I tried installing the program in anaconda and manual but both shows the same error.

Here is my input for anphon. The total number of atoms is 128. Would the calculation with more memory solve the error?

&general
 PREFIX = LPS_scph2-2
 MODE = SCPH
 NKD = 3; KD = Li P S
 FCSXML = LPS_anharmonic.xml
 NONANALYTIC = 0
 TMIN =0; TMAX = 1000; DT = 50
 DELTA_E = 5.0
/

&cell
  23.7264589626202
  1.0  0.0   0.0
  0.0  1.06247  0.0
  0.0  0.0   1.2519462
/

&kpoint
2
3 3 2
/

&scph
 SELF_OFFDIAG = 0
 IALGO = 1
 MAXITER = 2000
 MIXALPHA = 0.05
 LOWER_TEMP = 1
 KMESH_INTERPOLATE = 3 3 2
 KMESH_SCPH = 3 3 2
 TOL_SCPH = 1.0e-7
/

**Comment by ttadano** [comment]
Can you tell us the following information?

- version of alamode
- used compilers and MPI library
- detailed error message

**Reply by umichwbaek** [reply]
I directly installed the latest version from git clone, so 1.5.0 and develop branch.

I tried the following compilers and libraries
1. mpicxx and oneapi (HPC)
2. mpicc and gcc/openmpi (HPC)
3. mpicxx and libgomp (all compilers and libraries from conda install on my local computer)

but all crashes postprocessing step of SCPH/QHA.

The output stops like above what I showed on my first post and error message shows just like simply

free(): invalid pointer

without any more lines

or

[gl3166:2713995:0:2713995] Caught signal 11 (Segmentation fault: address not mapped to object at address (nil))
anphon: malloc.c:4106: _int_malloc: Assertion `(unsigned long) (size) >= (unsigned long) (nb)' failed.
[gl3166:2713995] *** Process received signal ***
[gl3166:2713995] Signal: Aborted (6)
[gl3166:2713995] Signal code:  (-6)
[gl3166:2713995] [ 0] /lib64/libpthread.so.0(+0x12cf0)[0x151f7c1bdcf0]
[gl3166:2713995] [ 1] /lib64/libc.so.6(gsignal+0x10f)[0x151f7a765acf]
[gl3166:2713995] [ 2] /lib64/libc.so.6(abort+0x127)[0x151f7a738ea5]
[gl3166:2713995] [ 3] /lib64/libc.so.6(+0x971aa)[0x151f7a7ae1aa]
[gl3166:2713995] [ 4] /lib64/libc.so.6(+0x9a949)[0x151f7a7b1949]
[gl3166:2713995] [ 5] /lib64/libc.so.6(__libc_malloc+0x1f2)[0x151f7a7b2982]
[gl3166:2713995] [ 6] /lib64/libucs.so.0(objalloc_create+0x21)[0x151f7928ae21]
[gl3166:2713995] [ 7] /lib64/libucs.so.0(+0x7ca73)[0x151f791d2a73]
[gl3166:2713995] [ 8] /lib64/libucs.so.0(+0x7e88c)[0x151f791d488c]
[gl3166:2713995] [ 9] /lib64/libucs.so.0(+0x7ea0c)[0x151f791d4a0c]
[gl3166:2713995] [10] /lib64/libucs.so.0(+0x619ce)[0x151f791b79ce]
[gl3166:2713995] [11] /lib64/libucs.so.0(+0x62228)[0x151f791b8228]
[gl3166:2713995] [12] /lib64/libucs.so.0(ucs_debug_backtrace_create+0x50)[0x151f791b84b0]
[gl3166:2713995] [13] /lib64/libucs.so.0(+0x62a14)[0x151f791b8a14]
[gl3166:2713995] [14] /lib64/libucs.so.0(ucs_handle_error+0x2e0)[0x151f791bb150]
[gl3166:2713995] [15] /lib64/libucs.so.0(+0x6533c)[0x151f791bb33c]
[gl3166:2713995] [16] /lib64/libucs.so.0(+0x6550a)[0x151f791bb50a]
[gl3166:2713995] [17] /lib64/libpthread.so.0(+0x12cf0)[0x151f7c1bdcf0]
[gl3166:2713995] [18] /home/wbaek/software/alamode/build/anphon/anphon(_ZNK7PHON_NS11Integration23calc_weight_tetrah
edronEjPKjPKddjPKS2_Pd+0xbd)[0x9123ff]
[gl3166:2713995] [19] /home/wbaek/software/alamode/build/anphon/anphon[0x96ae93]
[gl3166:2713995] [20] /sw/pkgs/arc/gcc/10.3.0/lib64/libgomp.so.1(GOMP_parallel+0x42)[0x151f7b45d772]
[gl3166:2713995] [21] /home/wbaek/software/alamode/build/anphon/anphon(_ZNK7PHON_NS3Dos8calc_dosEjjPKjPKPKdjRKSt6vec
torIdSaIdEEjijPKS2_Pd+0x96)[0x968a9e]
[gl3166:2713995] [22] /home/wbaek/software/alamode/build/anphon/anphon(_ZNK7PHON_NS3Dos29calc_dos_from_given_frequen
cyEPKNS_17KpointMeshUniformEPKPKdjPKPKjPd+0x159)[0x969cdf]
[gl3166:2713995] [23] /home/wbaek/software/alamode/build/anphon/anphon(_ZN7PHON_NS4Scph11postprocessEPPPPSt7complexI
dES6_S6_PKNS_17KpointMeshUniformEPPPNS_15MinimumDistListEbi+0xb88)[0x9b5316]
[gl3166:2713995] [24] /home/wbaek/software/alamode/build/anphon/anphon(_ZN7PHON_NS4Scph9exec_scphEv+0x8f1)[0x9b46ab]
[gl3166:2713995] [25] /home/wbaek/software/alamode/build/anphon/anphon(_ZNK7PHON_NS4PHON30execute_self_consistent_ph
ononEv+0x333)[0x973c63]
[gl3166:2713995] [26] /home/wbaek/software/alamode/build/anphon/anphon(_ZN7PHON_NS4PHONC1EiPPcP19ompi_communicator_t
+0x34c)[0x9726fa]
[gl3166:2713995] [27] /home/wbaek/software/alamode/build/anphon/anphon(main+0x48)[0x939131]
[gl3166:2713995] [28] /lib64/libc.so.6(__libc_start_main+0xe5)[0x151f7a751d85]
[gl3166:2713995] [29] /home/wbaek/software/alamode/build/anphon/anphon(_start+0x2e)[0x824d1e]
[gl3166:2713995] *** End of error message ***


depending on the number of nodes and DELTA_E value.

**Reply by ttadano** [reply]
Thank you for the report. I tested the codes with my inputs but did not reproduce the error. 

Can you tell us the following?

- Does the same error occur when you run `anphon` without `mpirun` option?
- Does setting `ISMEAR=0` solve the issue?
